src/context/ActiveTab.js

```diff
@@ -0,0 +1,62 @@
+// @flow
+/* eslint-disable react/no-unused-state */
+
+import * as React from 'react';
+
+export type TabType = 'search' | 'summary' | 'account' | null;
+type Props = {|
+  children: React.Node,
+|};
+type StateValues = {
+  activeTab: TabType,
+};
+
+type StateCallbacks = {
+  activateTab: (tab: TabType) => void,
+  toggleTab: (tab: TabType) => void,
+};
+
+export type ActiveTabType = StateValues & StateCallbacks;
+
+const initialState: StateValues = {
+  activeTab: null,
+};
+
+export const ActiveTab = React.createContext({
+  ...initialState,
+  activateTab: (tab: TabType) => null, // eslint-disable-line no-unused-vars
+  toggleTab: (tab: TabType) => null, // eslint-disable-line no-unused-vars
+});
+
+class ActiveTabProvider extends React.Component<Props, ActiveTabType> {
+  constructor(props: Props) {
+    super(props);
+
+    const activateTab = (tab: TabType) => {
+      this.setState({ activeTab: tab });
+    };
+
+    const toggleTab = (tab: TabType) => {
+      this.setState(prevState => ({
+        activeTab: prevState.activeTab === tab ? null : tab,
+      }));
+      this.forceUpdate();
```

Comment:
we should try to avoid all uses of `forceUpdate()`

Reply:
Agreed, but I didn't add that, I just moved it from another part of the code. We should remove it, but maybe this PR is big enough already :-)